template_feed/Microsoft.DotNet.Common.ItemTemplates/content/BuildProps/Directory.Build.props

```diff
@@ -0,0 +1,10 @@
+<Project>
+  <!-- See https://aka.ms/dotnet/msbuild/customize for more details on customizing your build -->
+<!--#if inherit -->
+  <Import Project="$(MSBuildThisFileDirectory)\..\Directory.Build.props"
+          Condition="Exists('$(MSBuildThisFileDirectory)\..\Directory.Build.props')" />
```

Comment:
This will not work well if the file is in the root directory of a drive. Wondering if there's a safer way to do this.

Reply:
Also - and apologies for nit-picking, I don't have nearly as much experience as you so feel free to ignore - wouldn't the developer typically want to chain to Directory.Build.props in *a* parent directory, not necessarily only one level above?

Reply:
@ladipro both are valid points - though I don't think folks will often be invoking the `buildprops` template with the `inherit` flag at such a location :) Regarding the second point, ideally I'd have some sense of where a parent D.B.p/t is and I'd be able to directly specify that. Maybe something like `inherit:2` or some other way to specify a depth? That felt a) hard and b) awkward, so I went with a very simple format. I mostly wanted to communicate to users that the mechanism is hierarchical and non-continuing (unlike .editorconfig).

Reply:
There's an idea - maybe D.B.p/t should implicitly load all matching D.B.p/t in parent directories up to some root? We'd need a way to mark the root easily/cheaply, though.

Reply:
If it implicitly imported all on the way to root, it would be susceptible to double importing if the developer is not careful.

I'd say ideally it should be robust so no matter where the developer adds the file, or later moves the file to (such as the root; conceivable in CI or containers?), it will keep the same (expected) behavior.

What do you think about getting the path using [GetPathOfFileAbove](https://learn.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#msbuild-getpathoffileabove)?

Reply:
oh now that's awesome. refactoring to use that immediately.

Reply:
Done, thank you for the reminder.